### PR TITLE
feat: deep dive topics proposal

### PR DESCRIPTION
**Pull request by alanshaw**
This is a proposal for deep dive topics, they've been sourced from various folks (you know who you are - thank you ❤️). This is a work in progress, but worth PR-ing for some feedback/review. There are a few that might need to be cut, I do not yet have an idea for an owner for _all_ of them but there's more here than we need so hopefully it should be ok!

If you have any additions to be considered, please comment here 🤗

### CODE REVIEW

**Comment by daviddias** [comment]
Awesome gathering @alanshaw!! 

Would you like to add the Deep Dive template here so that folks that are owning a deep dive can just c&p and create their one pager?

**Comment by daviddias** [comment]
Also, please deprecate the original doc as you move things here :)

**Review by momack2 — APPROVED** [review]

**Review by daviddias — APPROVED** [review]
Consider:
- Merging as is
- Creating the files for the one pagers right away
- Link the one pagers from the table

File: DEEP_DIVES/README.md [hunk 1]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
 ### How
 
 - **Duration**: 1:45h
+
+### Topics
+
+| # | Topic | Description |
+| :---: | --- | --- |
+| 1 | Private DHT |  |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:27]: 👍🏽

File: DEEP_DIVES/README.md [hunk 2]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+
+### Topics
+
+| # | Topic | Description |
+| :---: | --- | --- |
+| 1 | Private DHT |  |
+| 2 | Scaling up the DHT 📦 | What can we do to optimise the performance of our DHT? Are there alternatives? |
+| 3 | Reducing the information leakage on a libp2p dial |  |
+| 4 | Locking the IPFS repo for concurrent access | How do we manage concurrent access to an IPFS repo when we have multiple tabs on the same domain or web workers or service workers? How do we avoid being too restrictive (i.e. locking everything) whilst guarding against race conditions and concurrency problems? |
+| 5 | Identity for DApps 📦 | What are the next challenges post IDM? |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:31]: 👍🏽

File: DEEP_DIVES/README.md [hunk 3]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 6 | Live Streaming Capabilities through IPFS to X thousand | |
+| 7 | Upgrading the HTTP-API to a RPC API | We desperately need a new remote API https://github.com/ipfs/http-api-spec/issues/116 |
+| 8 | Offline First | How can IPFS currently be used to support communities with little to no network bandwidth? What improvements to the protocol would make IPFS offer better solutions to these challenges? |
+| 9 | Package Managers: What Now, What Next? 📦 | Community evaluation and ideation session to recap the current (as of late June) state of the IPFS-enabled package manager landscape, discuss happiness and pain points, and do some ideation/co-creation  around the direction of our next rounds of package manager efforts. |
+| 10 | IPFS on Mobile |  |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:36]: 👍🏽

File: DEEP_DIVES/README.md [hunk 4]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| # | Topic | Description |
+| :---: | --- | --- |
+| 1 | Private DHT |  |
+| 2 | Scaling up the DHT 📦 | What can we do to optimise the performance of our DHT? Are there alternatives? |
+| 3 | Reducing the information leakage on a libp2p dial |  |
+| 4 | Locking the IPFS repo for concurrent access | How do we manage concurrent access to an IPFS repo when we have multiple tabs on the same domain or web workers or service workers? How do we avoid being too restrictive (i.e. locking everything) whilst guarding against race conditions and concurrency problems? |
+| 5 | Identity for DApps 📦 | What are the next challenges post IDM? |
+| 6 | Live Streaming Capabilities through IPFS to X thousand | |
+| 7 | Upgrading the HTTP-API to a RPC API | We desperately need a new remote API https://github.com/ipfs/http-api-spec/issues/116 |
+| 8 | Offline First | How can IPFS currently be used to support communities with little to no network bandwidth? What improvements to the protocol would make IPFS offer better solutions to these challenges? |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:34]: 👍🏽

File: DEEP_DIVES/README.md [hunk 5]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 10 | IPFS on Mobile |  |
+| 11 | Private content on IPFS 📦 | |
+| 12 | Building the IPFS Community | Community, evangelism, engagement, on ramp. |
+| 13 | IPFS for websites (static, dynamic, search, etc) | What are the best practices and open questions for using IPFS for static / dynamic websites? What about indexing and searching? |
+| 14 | IPFS in blockchain | |
+| 15 | Open research problems in IPFS | Content routing |
+| 16 | Revocation/Rotating of IPNS Keys | The idea of using a public key to identify mutable data is cool, but what happens when that key becomes compromised? |
+| 17 | Building Reusable Data Structures | IPLD and IPFS allow data to be portable and reused between applications. What are some good behaviors or work flows to promote data reuse instead of re-creation. |
+| 18 | Service and Content Discovery under Partition 📦 | We tend to think of our systems as operating in either the “global internet” or a “local network”, but the reality can be more complicated than that. Do our systems meet that challenge and if not what would we like them to do? |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:44]: 👍🏽

File: DEEP_DIVES/README.md [hunk 6]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 19 | Referencing HTTP/centralized web content from IPFS 📦 | Currently data is available to the IPFS network when a node advertises that they have data and provides it upon request. Would it be useful if we could advertise data provided by HTTP? What are the tradeoffs and mitigations we would need? |
+| 20 | How can IPFS support sharding 📦 | How to best pin an arbitrary part of a DAG https://github.com/ipfs/go-ipfs/pull/5142 |
+| 21 | A better HTTP API | Upgrading the IPFS HTTP API to be more RESTful and easier to use |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:47]: 👍🏽

File: DEEP_DIVES/README.md [hunk 7]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 22 | Garbage Collection how to make more efficient large block stores 📦 | We currently stop-the-world and collect, what's involved with incremental, generational. Could we use a separate datastore, something else? |
+| 23 | Connection prioritisation | How can we prioritise connections so that important connections are not dropped and important dials happen more quickly. `ipfs swarm friend add <addr> - stay connected as long as possible, don't kill connections, reconnect on failure`. _Right now_ we add the cluster nodes to the gateway nodes (`swarm connect`) every minute. |
+| 24 | Replication protocol 📦 | How can we more quickly sync a DAG or part of a DAG between two nodes? A compliment to bitswap, affectionately overloaded as GraphSync |
+| 25 | Do not replicate lists - DNR | How to stop your node from replicating content due to take down notices or other reasons |
+| 26 | Fast IPNS 📦 | How to increase the IPNS resolution speed |
+| 27 | MultiWriter IPNS 📦 | Allowing multiple peers to update an IPNS record |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:53]: 👍🏽

File: DEEP_DIVES/README.md [hunk 8]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 20 | How can IPFS support sharding 📦 | How to best pin an arbitrary part of a DAG https://github.com/ipfs/go-ipfs/pull/5142 |
+| 21 | A better HTTP API | Upgrading the IPFS HTTP API to be more RESTful and easier to use |
+| 22 | Garbage Collection how to make more efficient large block stores 📦 | We currently stop-the-world and collect, what's involved with incremental, generational. Could we use a separate datastore, something else? |
+| 23 | Connection prioritisation | How can we prioritise connections so that important connections are not dropped and important dials happen more quickly. `ipfs swarm friend add <addr> - stay connected as long as possible, don't kill connections, reconnect on failure`. _Right now_ we add the cluster nodes to the gateway nodes (`swarm connect`) every minute. |
+| 24 | Replication protocol 📦 | How can we more quickly sync a DAG or part of a DAG between two nodes? A compliment to bitswap, affectionately overloaded as GraphSync |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:50]: 👍🏽

File: DEEP_DIVES/README.md [hunk 9]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 25 | Do not replicate lists - DNR | How to stop your node from replicating content due to take down notices or other reasons |
+| 26 | Fast IPNS 📦 | How to increase the IPNS resolution speed |
+| 27 | MultiWriter IPNS 📦 | Allowing multiple peers to update an IPNS record |
+| 28 | WASM in JS IPFS | Should more WASM be used in JS IPFS, what for and where? |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:54]: 👍🏽

File: DEEP_DIVES/README.md [hunk 10]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 29 | Resource utilization | What are the different places an IPFS node will operate and what are the appropriate resource utilization profiles for them? |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:55]: 👍🏽

File: DEEP_DIVES/README.md [hunk 11]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 34 | IPNS link rot | Right now, IPNS needs to be refreshed by original publisher every day, otherwise it stops working. There should be a way to pin IPNS address so that other peers keep refreshing the last valid entry when original node goes offline, keeping the last version alive even if the original publisher disappears forever. |
+| 35 | Deterministic CIDs! Reproducible File Imports! Verifable HTTP Gateways! | How to deterministically get the same CID for the same data, even if ipfs.add defaults change? How can we validate data read via HTTP Gateway without running full IPFS node? https://github.com/ipfs/roadmap/issues/19#issuecomment-494529255 https://github.com/ipfs/in-web-browsers/issues/128 |
+| 36 | Key Management | How it works today? Where we need to be? Eg: `ipfs key` is missing primitives for signing and verifying things using keys, we dont have interop for key import/export, keys are stored unencrypted on disk etc. I see this as "where we are"+"what devs need" → "where we need to be for 1.0"  type of dive, producing some tangible AIs |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:62]: 👍🏽

File: DEEP_DIVES/README.md [hunk 12]
```diff
@@ -19,3 +19,76 @@ These design sessions are not about explaining how things work, they were focuse
...
+| 41 | Better NAT traversal so that Relay servers are a last (not first) resort | The current cost of running relay servers for IPFS is unsustainable.  We need to implement better NAT hole punching mechanisms in libp2p to reduce the usage of relay servers by NAT’d hosts.  TCP hole punching has a success rate of ~60%, so we should be preferentially hole punching.  TCP simultaneous open is one example.  STUN servers are another example; yet another is making every non-NATed host in a libp2p network a relay node. |
+| 42 | libp2p connection bootstrapping and protocol negotiation 2.0 | Our connection bootstrapping process is chatty and inefficient. We incur in several round trips upfront just to negotiate the basics (encryption, muxing). Stream-level protocol negotiation is also inefficient. We are working on Multiselect 2.0 and a revamped connection bootstrapping process. This is the very heart of libp2p. Help us get this right. |
```
Comment (daviddias) [inline DEEP_DIVES/README.md:68]: 👍🏽

**Review by daviddias — APPROVED** [review]
Added some 👍🏽 to help sort and filter the final top 30 :)

**Comment by daviddias** [comment]
I'm merging this one for now with the blessing of @alanshaw :)

**Pull request merged**